ethereum_bridge/src/parameters.rs

```diff
@@ -157,13 +174,78 @@ impl EthereumBridgeConfig {
         // Initialize the storage for the Bridge Pool VP.
         bridge_pool_vp::init_storage(storage);
     }
+
+    /// Reads the latest [`EthereumBridgeConfig`] from storage. If it is not
+    /// present, `None` will be returned - this could be the case if the bridge
+    /// has not been bootstrapped yet. Panics if the storage appears to be
+    /// corrupt.
+    pub fn read<DB, H>(storage: &Storage<DB, H>) -> Option<Self>
+    where
+        DB: storage::DB + for<'iter> storage::DBIter<'iter>,
+        H: storage::traits::StorageHasher,
+    {
+        let min_confirmations_key = bridge_storage::min_confirmations_key();
+        let native_erc20_key = bridge_storage::native_erc20_key();
+        let bridge_contract_key = bridge_storage::bridge_contract_key();
+        let governance_contract_key = bridge_storage::governance_contract_key();
+
+        let Some(min_confirmations) = StorageRead::read::<MinimumConfirmations>(
+            storage,
+            &min_confirmations_key,
+        )
+        .unwrap_or_else(|err| {
+            panic!("Could not Borsh-deserialize {min_confirmations_key}: {err:?}")
+        }) else {
```

Comment:
I think the first error you encounter is an I/O error, no?

Reply:
I'm not sure, you may be right though, just going by the API of `StorageRead::read`, it doesn't specify the causes of errors that might be returned (beyond the wrapper types at https://github.com/anoma/namada/blob/1f8fd9d751e5754eb39018fdc2fd1a7ef7e71b37/core/src/ledger/storage_api/error.rs#L6-L15), so I've made this error message more generic (the `{err:?}` should give more context anyway).